x-pack/filebeat/input/gcs/state.go

```diff
@@ -27,32 +27,41 @@ type Checkpoint struct {
 	// name of the latest blob in alphabetical order
 	ObjectName string
 	// timestamp to denote which is the latest blob
-	LatestEntryTime *time.Time
+	LatestEntryTime time.Time
 	// list of failed jobs due to unexpected errors/download errors
 	FailedJobs map[string]int
+	// map to contain root type information for json objects
+	IsRootArray map[string]bool
+	// map to contain offset data
+	// if isRootArray == true for object, then PartiallyProcessed will treat offset as an array index
```

Comment:
I think these comments could be clearer (left unfinished because I don't think I 100% understand what you are intending them to mean).

- `IsRootArray`: a mapping from object name to whether the object is ...
- `PartiallyProcessed`: a mapping from object name to array index ...

`PartiallyProcessed` sounds like a boolean to me, so I think the name is probably not right; `LastProcessedOffset`?